### ISSUE TITLE
`python_use_unqualified_type_names = True` ignored in base classes

### ISSUE THREAD

**Issue by ktbarrett** [sphinx-doc/sphinx#10027]
### Describe the bug

`python_use_unqualified_type_names = True` should print unqualified names if they are documented. However when using the option `:show-inheritance:` with `:autoclass:` names that are documented, and link correctly, still have the fully-qualified name printed.

### How to Reproduce

```
$ git clone https://github.com/ktbarrett/pyhdltypes/tree/sphinx-error
$ cd pyhdltypes
$ nox -e docs
$ # open docs_out/reference.html#logic-array-types
```

See `LogicArrayBase`, `StdLogicArray`, `X01ZArray`, and `BitArray`.

### Expected behavior

I expect the types that are documented to be printed with unqualified names per the intent of the option `python_use_unqualified_type_names`.

### Your project

https://github.com/ktbarrett/pyhdltypes/tree/sphinx-error

### Screenshots

![image](https://user-images.githubusercontent.com/22162772/147613183-cb179eed-26b6-46c5-bf80-c58100c18bb4.png)

All the listed base types are documented and the links work, but the names are fully qualified when they shouldn't be.

### OS

Linux

### Python version

3.9

### Sphinx version

4.3.2

### Sphinx extensions

autodoc

**Comment by tk0miya** [comment]
`python_use_unqualified_type_names` is not related to this case at all because it does not control the behavior of autodoc. It should be changed by `autodoc_typehints_format='short'` that will be added to v4.4.0.

On the other hand, it does not work with `:show-inheritance:` option. So I posted a fix to that in #10034. Please try it.

**Comment by ktbarrett** [comment]
@tk0miya Yes, the new option and the fix in #10034 do work.
![image](https://user-images.githubusercontent.com/22162772/147861617-42192ad9-8192-4d46-b47e-36c5a9cc895a.png)

**Comment by tk0miya** [comment]
Thank you for confirming. Just merged. It will be released as 4.4.0 soon (maybe this weekend).
